client/lib/create-selector/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import memoize from 'lodash/memoize';
+import shallowEqual from 'react-pure-render/shallowEqual';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+/**
+ * At runtime, assigns a function which returns a cache key for the memoized
+ * selector function, given a state object and a variable set of arguments. In
+ * development mode, this warns when the memoized selector is passed a complex
+ * object argument, as these cannot be depended upon as reliable cache keys.
+ *
+ * @type {Function} Function returning cache key for memoized selector
+ */
+const getCacheKey = ( () => {
+	if ( 'development' !== config( 'env' ) ) {
+		return ( state, ...args ) => args.join();
+	}
+
+	const warn = require( 'lib/warn' );
+	return ( state, ...args ) => {
+		const hasInvalidArg = args.some( ( arg ) => {
+			return arg && -1 === [ 'number', 'boolean', 'string' ].indexOf( typeof arg );
```

Comment:
>Not sure if it's worth bringing in includes, but maybe:

Sure. The rebased d3d44720f1ade9af26cba7a9956434d58eaf6f1e uses `lodash/includes` and assigns valid types to a constant value top-of-file.